pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,6 +94,34 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	var contents bytes.Buffer
```

Comment:
It feels kind of weird to pass in a buffer to be filled inside `createSecrets`. Why don't we just initialize the buffer inside createSecrets instead of passing it in?

Reply:
Good point